src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/VersionFiles.cs

```diff
@@ -76,5 +79,63 @@ public static string CalculateDotnetToolsJsonElementName(string dependencyName)
         {
             return dependencyName;
         }
+
+        /// <summary>
+        /// Reverse a version in the Arcade style (https://github.com/dotnet/arcade/blob/fb92b14d8cd07cf44f8f7eefa8ac58d7ffd05f3f/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.BeforeCommonTargets.targets#L18)
+        /// back to an OfficialBuildId + ReleaseLabel which we can then supply to get the same resulting version number.
+        /// </summary>
+        /// <param name="repoName">The source build name of the repo to get the version info for.</param>
+        /// <param name="version">The complete version, e.g. 1.0.0-beta1-19720.5</param>
+        public static (string BuildId, string ReleaseLabel) DeriveBuildInfo(string repoName, string version)
+        {
+            var nugetVersion = new NuGetVersion(version);
+
+            if (string.IsNullOrWhiteSpace(nugetVersion.Release))
+            {
+                // finalized version number (x.y.z) - probably not our code
+                // Application Insights, Newtonsoft.Json do this
+                return (DateTime.Now.ToString("yyyyMMdd.1"), string.Empty);
+            }
+
+            var releaseParts = nugetVersion.Release.Split('-', '.');
+            if (repoName.Contains("nuget"))
+            {
+                // NuGet does this - arbitrary build IDs
+                return (DateTime.Now.ToString("yyyyMMdd.1"), releaseParts[0]);
+            }
+
+            if (releaseParts.Length == 3)
+            {
+                // VSTest uses full dates for the first part of their preview build numbers
+                if (repoName.Contains("vstest"))
+                {
+                    return ($"{releaseParts[1]}.{releaseParts[2]}", releaseParts[0]);
+                }
+                else if (int.TryParse(releaseParts[1], out int datePart) && int.TryParse(releaseParts[2], out int buildPart))
+                {
+                    if (datePart > 1 && datePart < 8 && buildPart > 1000 && buildPart < 10000)
+                    {
+                        return (releaseParts[2], $"{releaseParts[0]}.{releaseParts[1]}");
+                    }
+                    else
```

Comment:
There is a return in the above branch, the else is not needed here.

Reply:
Ah, missed this when porting from https://github.com/dotnet/installer/blob/9ec5d7f1bc4202809d4c59407daead9c0c5df0c9/src/SourceBuild/Arcade/src/Tarball_WriteSourceRepoProperties.cs#L99